core/pkg/eval/json_evaluator.go

```diff
@@ -122,6 +122,11 @@ func (je *JSONEvaluator) ResolveAllValues(ctx context.Context, reqID string, con
 	var err error
 	allFlags := je.store.GetAll()
 	for flagKey, flag := range allFlags {
+		if flag.State == Disabled {
```

Comment:
What do you think about filtering out disabled flags when we're unmarshing the flag config? That would also indirectly resolve [this issue](https://github.com/open-feature/flagd/pull/670).

Reply:
:thinking:

Reply:
I think it's important for disabled flags to act as if they don't exist, as I mentioned elsewhere... how exactly we do that doesn't matter as much to me.

Reply:
shall we do the following, let's agree on the fixes proposed through #670 & #672 . And then follow up on the proposed behavior change - `Ignore/not store Disabled flags as if they do not exist`

The reason is this change requires considerable amount of changes. Besides, the above fixes do not add new behavior and merely bug fixes (can go into a release)

Reply:
Created issue https://github.com/open-feature/flagd/issues/673 , I can work on this. Feel free to update/comment on the issue